usecase/pokemon/pokemon.go

```diff
@@ -0,0 +1,72 @@
+package pokemon
+
+import (
+	"io"	
+	"bootcamp/domain/model"
+	"bootcamp/service/db"
+	"bootcamp/utils"
+)
+
+/*
+GetPokemon retrieve all existent Pokemon from the database
+*/
+func GetPokemon() (model.PokemonList, error) {
+	pokemonList, err := db.GetPokemon()
+	return pokemonList, err
+}
+
+/*
+GetPokemonById retrieves Pokemon information that matches with a given id from the database
+*/
+func GetPokemonById(params map[string]string) (model.Pokemon, error) {
+	var pokemon model.Pokemon
+	objectId, err := utils.GetObjectIdFromParams(params)
+
+	if err == nil {
+		pokemon, err = db.GetPokemonById(objectId)
+	}
+
+	return pokemon, err
+}
+
+/*
+AddPokemon inserts Pokemon information in the database
+*/
+func AddPokemon(reader io.ReadCloser) (model.Pokemon, error) {
```

Comment:
Just as a reminder to avoid passing Request data into usecases. Get that in the controller, convert that into an internal data type(like a struct with parsed attributes from the request body) and hand it over to the usecase.

Reply:
*